### ISSUE TITLE
Firebase update breaks ChatSDK

### ISSUE THREAD

**Issue by MartPiet** [chat-sdk/chat-sdk-ios#464]
1. **Is the bug present in the demo Chat SDK project?**
- Unrelated

2. **What modifications have you made to the Chat SDK?**
- UI changes
- Added property into the db
- Consistent thread creation

I think all of these changes are unrelated to this dependency issue.

3. **iOS Version? Device or simulator?**
iOS 14, Simulator & Device

4. **Steps taken to reproduce the problem:**
Update Firebase Pods (pod update), Changelog regarding the versioning:

```
Installing Firebase 7.1.0 (was 6.22.0)
Installing FirebaseAnalytics 7.1.0 (was 6.4.1)
Installing FirebaseAuth 7.1.0 (was 6.5.1)
Installing FirebaseCore 7.1.0 (was 6.6.6)
Installing FirebaseCoreDiagnostics 7.1.0 (was 1.2.3)
Installing FirebaseCrashlytics 7.1.0 (was 4.0.0-beta.7)
Installing FirebaseDatabase 7.1.0 (was 6.1.4)
Installing FirebaseFunctions 7.1.0 (was 2.5.1)
Installing FirebaseInstallations 7.1.0 (was 1.1.1)
Installing FirebaseInstanceID 7.1.0 (was 4.3.3)
Installing FirebaseMessaging 7.1.0 (was 4.3.0)
Installing FirebaseStorage 7.1.0 (was 3.6.0)
Installing GTMSessionFetcher 1.5.0 (was 1.3.1)
Installing GoogleAppMeasurement 7.1.0 (was 6.4.1)
Installing GoogleDataTransport 8.0.1 (was 5.1.1)
Installing GoogleUtilities 7.1.0 (was 6.5.2)
Installing PromisesObjC 1.2.11 (was 1.2.8)
Installing nanopb 2.30906.0 (was 0.3.9011)
Removing FirebaseAnalyticsInterop
Removing FirebaseAuthInterop
Removing FirebaseCoreDiagnosticsInterop
Removing GoogleDataTransportCCTSupport
Removing Protobuf
```

5. **Expected result:**
Complied App.

6. **Actual result:**
Compiling fails in Firebase+Paths.h, line 13:
"Cannot define category for undefined class 'FIRDatabaseReference'"

7. **Comments:**
These ChatSDK versions are used:

```
- ChatSDK (4.13.1):
    - ChatSDK/Complete (= 4.13.1)
  - ChatSDK/Complete (4.13.1):
    - ChatSDK/Reachability
    - ChatSDK/UI
  - ChatSDK/Core (4.13.1):
    - AFNetworking/NSURLSession (~> 3.2.1)
    - DateTools (~> 2.0)
    - RXPromise (~> 1.0)
    - SAMKeychain
  - ChatSDK/CoreData (4.13.1):
    - ChatSDK/Core
  - ChatSDK/Reachability (4.13.1):
    - ChatSDK/Core
    - Reachability
  - ChatSDK/UI (4.13.1):
    - ChatSDK/CoreData
    - ChatSDKKeepLayout
    - CountryPicker (~> 1.0)
    - Hakawai (~> 5.1.5)
    - MBProgressHUD (~> 1.0)
    - SDWebImage (~> 5.0)
    - StaticDataTableViewController (~> 2.0)
    - Toast (~> 4.0.0)
    - TOCropViewController (~> 2.0)
    - VENTokenField (~> 2.0)
  - ChatSDKFirebase/Adapter (4.13.1):
    - ChatSDK
    - Firebase/Auth
    - Firebase/Database
  - ChatSDKFirebase/FileStorage (4.13.1):
    - ChatSDKFirebase/Adapter
    - Firebase/Storage
  - ChatSDKFirebase/Push (4.13.1):
    - ChatSDKFirebase/Adapter
    - Firebase/Functions
    - Firebase/Messaging
  - ChatSDKKeepLayout (1.7.7):
    - ChatSDKKeepLayout/Swift (= 1.7.7)
  - ChatSDKKeepLayout/ObjC (1.7.7)
  - ChatSDKKeepLayout/Swift (1.7.7):
    - ChatSDKKeepLayout/ObjC
```

To sum up: Updating firebase due to a deprecated Crashlytics version leads to breaking the dependencies for ChatSDK. Newer ChatSDK versions do not fix this issue. I currently have no idea what to do to fix this since all this seems to be a library issue. 
The podspec of the ChatSDK does not have any explicit dependency versioning for firebase dependencies so I assume normally the ChatSDK should support the newer firebase versions.

**Comment by bensmiley** [comment]
I did update to the latest version of Firebase and it works fine using the latest version of Firebase. You can update using Cocoapods to 4.15.0.
